lib/receipt.js

```diff
@@ -71,15 +74,16 @@ class Receipt {
     }
 
     /**
-     * Will hash and sign the receipt given a private key
-     * @param {String|PrivateKey} privateKey - Operator private key
+     * Will hash and sign the receipt with the wallet passed into the constructor
      */
-    sign(privateKey) {
+    async sign() {
         const h = hashReceipt(this.toJSON());
-        const s = sign(h, privateKey);
+        const hArr = ethers.utils.arrayify(h);
```

Comment:
Instead of having to repeat this all over the place, why not move it into the "signMessage()" method in Wallet?

Reply:
Idea is to avoid changing the behaviour of internal ethers Wallet functions, which could break other internal operations